File: Sources/RakaatCore/Posture.swift
import Foundation

/// The salah postures we classify each frame into.
public enum Posture: String, Equatable, Sendable, CaseIterable {
    case standing   // qiyam / i'tidal — torso upright, hips high above ankles
    case ruku       // bowing — torso ~horizontal, head still well above floor
    case sujud      // prostration — torso low, head near the floor
    case jalsa      // sitting — torso upright, hips low near ankles
    case unknown    // not enough joints to decide

    /// Human-readable Indonesian label for the live overlay.
    public var labelID: String {
        switch self {
        case .standing: return "BERDIRI"
        case .ruku: return "RUKU"
        case .sujud: return "SUJUD"
        case .jalsa: return "DUDUK"
        case .unknown: return "—"
        }
    }
}

File: web/src/core/posture.ts
export enum Posture {
  Standing = "standing", // qiyam / i'tidal — torso upright, hips high
  Ruku = "ruku", // bowing — torso ~horizontal, head still high
  Sujud = "sujud", // prostration — torso low, head near floor
  Jalsa = "jalsa", // sitting — torso upright, hips low
  Unknown = "unknown", // not enough joints
}

/** Indonesian label for the live overlay. */
export function postureLabel(p: Posture): string {
  switch (p) {
    case Posture.Standing:
      return "BERDIRI";
    case Posture.Ruku:
      return "RUKU";
    case Posture.Sujud:
      return "SUJUD";
    case Posture.Jalsa:
      return "DUDUK";
    default:
      return "—";
  }
}
